### PR TITLE
jasper: update to 3.0.6

### PR DESCRIPTION
**Pull request by jmroot**
JasPer version 3 comes with an ABI break and also a handful of API changes (mostly minor, but changes nonetheless): https://jasper-software.github.io/jasper-manual/releases/version-3.0.2/html/index.html#overview__news

Hopefully CI will show what needs to be fixed.

### CODE REVIEW

**Comment by cgilles** [comment]
digiKam use Jasper toolkit to deal with JPEG2000. I'm in contact with Jasper lead developer about the new 3.x release.
Warning : API has changed and needs fixes from client code to run. It can compile but not run properly as well, depending of the features used from the library.
For digiKam a test plan is planed in the near future, when time permit. I recommend to prepare the update of libjasper port and wait a little bit to see if all is fine from client code.
Just my 10 cts.
Gilles Caulier

**Comment by jmroot** [comment]
> I recommend to prepare the update of libjasper port and wait a little bit to see if all is fine from client code.

Certainly; that's the purpose of this PR.

**Comment by jmroot** [comment]
These ports need build fixes due to jasper changes: exact-image, libdevil, wgrib2, [xv](https://github.com/jasper-software/xv)

These ports were already failing before this change: ResIL, gstreamer010-gst-plugins-bad, digikam, kdelibs4 (maybe?), kopete, metview

qt*-qtimageformats need to be tested manually because CI wants to build all the unchanged subports as well.

**Comment by mascguy** [comment]
Josh, pushed one additional commit, with two minor changes. (Purposely didn't squash the change, so that you can see what was updated.)

**Comment by mascguy** [comment]
Josh, I fixed the merge conflicts with `gdk-pixbuf2` and `gegl-devel` (both recently updated), along with bumping Jasper to 3.0.2.

**Comment by mascguy** [comment]
> Josh, I fixed the merge conflicts with `gdk-pixbuf2` and `gegl-devel` (both recently updated), along with bumping Jasper to 3.0.2.

But no worries if you'd prefer to stick with 3.0.1, and/or not include the minor portfile formatting changes.

**Review by mascguy — APPROVED** [review]

**Comment by mascguy** [comment]
It looks like we'll need to make an additional change, to fix builds for 10.7 through 10.9.

Blacklisting older Clang versions would likely take care of it. But alternatively, we might need to bump the cxx standard.

https://build.macports.org/builders/ports-10.9_x86_64-builder/builds/181045/steps/install-port/logs/stdio

**Comment by mascguy** [comment]
Rebased against master, and fixed merge conflicts.

File: graphics/jasper/Portfile [hunk 1]
```diff
@@ -41,6 +48,14 @@ variant docs description {build documentation} {
...
-    configure.args-append   -DJAS_ENABLE_DOC=true
+    configure.args-replace  -DJAS_ENABLE_DOC=OFF \
+                            -DJAS_ENABLE_DOC=ON
 }
+
+variant heif description {Enable the use of the HEIF Library} {
+    depends_lib-append      port:libheif
+    configure.args-replace  -DJAS_ENABLE_LIBHEIF=OFF \
+                            -DJAS_ENABLE_LIBHEIF=ON
+}
```
Comment (mascguy) [inline graphics/jasper/Portfile:59]: Josh, would you be interested in enabling experimental HEIC support too, when this variant is enabled?

Comment (mascguy) [reply graphics/jasper/Portfile:59]: > Josh, would you be interested in enabling experimental HEIC support too, via this variant?

If so, the change is available via `jasper-devel`, for testing/validation:

https://github.com/macports/macports-ports/commit/3b0b08aff32f2b15c014714e54ad2126f355b0af

**Comment by mascguy** [comment]
Rebased against master, and updated to 3.0.6.

**Comment by mascguy** [comment]
Rebased against master, to fix merge conflict.

**Comment by mascguy** [comment]
Port updated via commit:

https://github.com/macports/macports-ports/commit/5341ba92b2f5af75558379b8af93a65d4ce33f3a

**Pull request closed without merging**